src/checkpoint.hh

```diff
@@ -703,6 +730,55 @@ SUPPORTED_EXTERNAL_TYPES = {"Rope", "answer_pknot_mfe", "pktype",
       stream << indent() << "}" << endl;
       dec_indent();
       stream << indent() << "}" << endl << endl;
+      if (cyk) {
+        stream << indent() << "// recalculate the broken List_Ref's " << endl;
+        stream << indent() << "unsigned int i, j;" << endl;
+        stream << indent() << "// sort by vector idx to match cyk loop order"
+               << endl;
+        stream << indent() << "std::sort(broken_listrefs.begin(), "
+               << "broken_listrefs.end()," << endl
+               << indent() << "          [](const std::pair<int, size_t> &l,"
+               << "const std::pair<int, size_t> &r) {" << endl
+               << indent() << "          return l.second < r.second;});"
+               << endl;
+        stream << indent() << "for (const auto& [table_i, vec_i] : "
+               << "broken_listrefs) {" << endl;
+        inc_indent();
+        stream << indent() << "std::pair<int, int> indices = "
+               << "map_1d_to_2d(vec_i);" << endl;
+       stream << indent() << "i = indices.first; j = indices.second;" << endl;
+        stream << indent() << "switch (table_i) {" << endl;
+        inc_indent();
+        int c = 0;
+        for (auto i = tables.begin(); i != tables.end(); ++i) {
+          const std::string &table_name = i->first;
+          stream << indent() << "case " << c << " :" << endl;
+          inc_indent();
+          // get table configuration (quadratic, linear etc.)
```

Comment:
attention: CONST tables can have running indices. This is why Gregor checks for indices being deleted like here https://github.com/jlab/gapc/blob/4bf7b5061136becd2bff70b58c9f5fbc95867ad8/src/cpp.cc#L2041-L2046 I struggled with this for a while for outside computation and finally also added https://github.com/jlab/gapc/blob/9337c750cc42be4c2fddddf90c31891088f4e3c8/src/table.cc#L73-L75

Not sure if this applies here, just wanted to make sure you are aware of these edge cases